### PR TITLE
Added GPU related metrics/Codacy errors

### PR DESCRIPTION
**Pull request by atharv67**

### CODE REVIEW

**Review by atharv67 — COMMENTED** [review]
Reverting changes of disk and system and pushing

File: src/driver.py [hunk 1]
```diff
@@ -54,16 +60,17 @@ def CollectMetrics(obj: dict) -> bool:
...
-    version = config["version"]
-    client_json["metadata"] = {"version": version, "time": datetime.utcnow().isoformat() + "Z"}
     token = (config["auth"]["username"], config["auth"]["password"])
-
+    # client_id=str((gma()))
+    # client_json["id"]=client_id
     if not CollectMetrics(client_json):
         sys.exit(1)
-
+    tests.test_driver.test_json_validation()
```
Comment (melonpii) [inline src/driver.py]: remove this. tests should not be called from driver.py

File: src/driver.py [hunk 2]
```diff
@@ -54,16 +60,17 @@ def CollectMetrics(obj: dict) -> bool:
         config = yaml.safe_load(file)
 
     # collect meta-data fields
-    version = config["version"]
```
Comment (melonpii) [inline src/driver.py:57]: why is this removed?

File: src/driver.py [hunk 3]
```diff
@@ -1,14 +1,18 @@
...
 
 import json
-from src.disk import Disk
-from src.system import System
-from src.network import Network
 import os
 import sys
 import yaml
-from datetime import datetime
+#from datetime import datetime
```
Comment (melonpii) [inline src/driver.py]: this should not be removed

**Review by melonpii — CHANGES_REQUESTED** [review]
pls check this pr again. It will break things.

**Comment by atharv67** [comment]
> pls check this pr again. It will break things.

Working on it

File: src/driver.py (hunk 3 above)
Comment (atharv67) [reply src/driver.py:7]: Unused import

**Comment by atharv67** [comment]
client_json {
  "metadata": {
    "version": "1.0.0",
    "time": "2022-10-08T00:10:36.080860Z"
  },
  "disk": [
    {
      "name": "/dev/disk1",
      "type": "apfs",
      "total_size": 0.49,
      "used": 0.02,
      "free": 0.47,
      "percentage": 4.08
    },
    {
      "name": "/dev/disk3",
      "type": "apfs",
      "total_size": 228.27,
      "used": 91.21,
      "free": 137.07,
      "percentage": 39.96
    },
    {
      "name": "/dev/disk4",
      "type": "hfs",
      "total_size": 0.02,
      "used": 0.0,
      "free": 0.02,
      "percentage": 0.0
    }
  ],
  "agent": {
    "hostname": "Atharvs-Air.lan",
    "bios_uuid": "00000000-0000-0000-0000-4225bb577fea",
    "platform": "Darwin",
    "release": "21.6.0",
    "version": "Darwin Kernel Version 21.6.0: Sat Jun 18 17:05:47 PDT 2022; root:xnu-8020.140.41~1/RELEASE_ARM64_T8101",
    "machine": "arm64",
    "is64bit": true,
    "physical_cores": 8,
    "logical_cores": 8,
    "total_memory_gb": 8.0
  },
  "metrics": {
    "cpu_load_5min": 19.02,
    "cpu_load_15min": 20.93,
    "avail_memory_gb": 2.03,
    "boot_days_ago": 16
  },
  "network": {
    "mac_address": "42:25:BB:57:7F:EA",
    "ip_address": "192.168.1.221",
    "hostname": "Atharvs-Air.lan",
    "connection_status": true,
    "down_speed": 235.62,
    "up_speed": 12.69,
    "time_now": "20:10:36"
  },
  "gpu": null
}

This is the json finally created

File: src/driver.py [hunk 4]
```diff
@@ -22,48 +26,49 @@ def CollectMetrics(obj: dict) -> bool:
...
     version = config["version"]
     client_json["metadata"] = {"version": version, "time": datetime.utcnow().isoformat() + "Z"}
     token = (config["auth"]["username"], config["auth"]["password"])
-
     if not CollectMetrics(client_json):
         sys.exit(1)
-
+    client_json = json.dumps(client_json, indent=2)
+    print("final_json", client_json)
+    print("client_json",client_json)
```
Comment (melonpii) [inline src/driver.py]: remove once

**Review by melonpii — APPROVED** [review]

**Review by VSangarya — APPROVED** [review]

**Pull request merged**